packages/cli/src/commands/fetch.ts

```diff
@@ -236,115 +212,104 @@ const shouldRecommendAlignMode = async (
   )
 }
 
-export const command = (
-  workspaceDir: string,
-  force: boolean,
-  interactive: boolean,
-  telemetry: Telemetry,
-  output: CliOutput,
-  spinnerCreator: SpinnerCreator,
-  mode: nacl.RoutingMode,
-  shouldCalcTotalSize: boolean,
-  inputServices?: string[],
-  inputEnvironment?: string,
-  stateOnly = false,
-): CliCommand => ({
-  async execute(): Promise<CliExitCode> {
-    log.debug(`running fetch command on '${workspaceDir}' [force=${force}, interactive=${
-      interactive}, mode=${mode}], environment=${inputEnvironment}, services=${inputServices}`)
-
-    const cliTelemetry = getCliTelemetry(telemetry, 'fetch')
-    const { workspace, errored, stateRecencies } = await loadWorkspace(workspaceDir, output,
-      { force, printStateRecency: true, spinnerCreator, sessionEnv: inputEnvironment })
-    if (errored) {
-      cliTelemetry.failure()
-      return CliExitCode.AppError
-    }
-
-    let useAlignMode = false
-    if (!force && mode !== 'align' && await shouldRecommendAlignMode(workspace, stateRecencies, inputServices)) {
-      const userChoice = await getChangeToAlignAction(mode, output)
-      if (userChoice === 'cancel operation') {
-        log.info('Canceling operation based on user input')
-        return CliExitCode.UserInputError
-      }
-      if (userChoice === 'yes') {
-        log.info(`Changing fetch mode from '${mode}' to 'align' based on user input`)
-        useAlignMode = true
-      }
-      log.info('Not changing fetch mode based on user input')
-    }
-
-    return fetchCommand({
-      workspace,
-      force,
-      interactive,
-      cliTelemetry,
-      output,
-      fetch: apiFetch,
-      getApprovedChanges: cliGetApprovedChanges,
-      shouldUpdateConfig: cliShouldUpdateConfig,
-      inputServices,
-      mode: useAlignMode ? 'align' : mode,
-      shouldCalcTotalSize,
-      stateOnly,
-    })
-  },
-})
-
 type FetchArgs = {
   force: boolean
   interactive: boolean
   stateOnly: boolean
-} & FetchModeArgs & ServicesArgs & EnvironmentArgs
-type FetchParsedCliInput = ParsedCliInput<FetchArgs>
+  mode: nacl.RoutingMode
+} & ServicesArg & EnvArg
 
-const fetchBuilder = createCommandBuilder({
-  options: {
-    command: 'fetch',
+const action = async (
+  {
+    input: { force, interactive, stateOnly, services, env, mode },
+    cliTelemetry,
+    config,
+    output,
+    spinnerCreator,
+    workingDir = '.',
+  }: DefActionInput<FetchArgs>
+): Promise<CliExitCode> => {
+  log.debug(`running fetch command on '${workingDir}' [force=${force}, interactive=${
+    interactive}, mode=${mode}], environment=${env}, services=${services}`)
```

Comment:
nit
```suggestion
  log.debug(`running fetch command on '%s' %o`, workingDir, { force, interactive, stateOnly, services, env, mode })
```
or, instead of destructuring `input` in the parameters, do something like this:
```suggestion
  log.debug('running fetch command on %s with options: %o', workingDir, input)
  const { force, interactive, stateOnly, services, env, mode } = input
```